metricbeat/module/php_fpm/php_fpm.go

```diff
@@ -0,0 +1,103 @@
+package php_fpm
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/status"
+)
+
+// HostParser is used for parsing the configured php-fpm hosts.
+var HostParser = parse.URLHostParserBuilder{
+	DefaultScheme: defaultScheme,
+	DefaultPath:   defaultPath,
+}.Build()
+
+// StatsClient provides access to php-fpm stats api
+type StatsClient struct {
+	address  string
+	user     string
+	password string
+	http     *http.Client
+}
+
+// NewStatsClient creates a new StatsClient
+func NewStatsClient(m mb.BaseMetricSet, isFullStats bool) *StatsClient {
+	var address string
+	address = m.HostData().SanitizedURI + "?json"
+	if isFullStats {
+		address += "&full"
+	}
+	return &StatsClient{
+		address:  address,
+		user:     m.HostData().User,
+		password: m.HostData().Password,
+		http:     &http.Client{Timeout: m.Module().Config().Timeout},
+	}
+}
+
+// Fetch php-fpm stats
+func (c *StatsClient) Fetch() (io.ReadCloser, error) {
+	req, err := http.NewRequest("GET", c.address, nil)
+	if c.user != "" || c.password != "" {
+		req.SetBasicAuth(c.user, c.password)
+	}
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error making http request: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("HTTP error %d: %s", resp.StatusCode, resp.Status)
+	}
+
+	return resp.Body, nil
+}
+
+// PoolStats defines all stats fields from a php-fpm pool
+type PoolStats struct {
```

Comment:
Indeed the response is in JSON and also it came to me that there would be a better way of handling this situation. But I wonder what will happen to the field names itself, since php-fpm replies a JSON with whitespace in field names (e.g. `accepted conn`)

Mind pointing out an implementation that uses it?